pkg/client/crypto.go

```diff
@@ -0,0 +1,123 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+//
+// This file is part of the Storj client library.
+//
+// The Storj client library is free software: you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// The Storj client library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with The Storj client library.  If not, see
+// <http://www.gnu.org/licenses/>.
+
+package client
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/hmac"
+	"crypto/sha256"
+	"crypto/sha512"
+	"encoding/base64"
+	"encoding/hex"
+	"errors"
+	"fmt"
+
+	bip39 "github.com/tyler-smith/go-bip39"
+)
+
+const (
+	bucketNameMagic = "398734aab3c4c30c9f22590e83a95f7e43556a45fc2b3060e0c39fde31f50272"
+	gcmDigestSize   = 16
+	ivSize          = 32
+)
+
+var bucketMetaMagic = []byte{66, 150, 71, 16, 50, 114, 88, 160, 163, 35, 154, 65, 162, 213, 226, 215, 70, 138, 57, 61, 52, 19, 210, 170, 38, 164, 162, 200, 86, 201, 2, 81}
```

Comment:
for side by side reviews, word wrapping around 80 cols is super nice. not saying we need to make it policy, but reviews are much better if there's some sensible newline placement